sdk/logical/audit.go

```diff
@@ -60,31 +61,35 @@ func (l *LogInput) BexprDatum(namespace string) *LogInputBexpr {
 	}
 }
 
-// TODO: PW: godoc + tests
+// Clone will attempt to create a deep copy of the LogInput.
+// This is preferred over attempting to use other libraries such as copystructure.
+// Audit formatting methods which parse LogInput into an audit request/response
+// entry, call receivers on the LogInput struct to get their value. These values
+// would be lost using copystructure as it cannot copy unexported fields.
+// If the LogInput type or any of the subtypes referenced by LogInput fields are
+// changed, then the Clone methods
 func (l *LogInput) Clone() (*LogInput, error) {
-	// Auth
+	// Close Auth
 	auth, err := cloneAuth(l.Auth)
 	if err != nil {
 		return nil, err
 	}
 
-	// Request
+	// Close Request
 	req, err := cloneRequest(l.Request)
 	if err != nil {
 		return nil, err
 	}
 
-	// Response
+	// Close Response
```

Comment:
Same here.

Reply:
I clearly closed the comment with copy/paste. 😆